pkg/protocol/engine/ledger/ledger/ledger.go

```diff
@@ -131,7 +131,13 @@ func (l *Ledger) AttachTransaction(block *blocks.Block) (attachedTransaction mem
 	if signedTransaction, hasTransaction := block.SignedTransaction(); hasTransaction {
 		signedTransactionMetadata, err := l.memPool.AttachSignedTransaction(signedTransaction, signedTransaction.Transaction, block.ID())
 		if err != nil {
-			l.retainTransactionFailure(block.ID(), err)
+			transactionID, txIDErr := signedTransaction.Transaction.ID()
+			if txIDErr != nil {
+				l.errorHandler(err)
```

Comment:
```suggestion
				l.errorHandler(txIDErr)
```

Shouldn't we pass the txIDErr? Or can you document in a comment why you pass `err` instead of `txIDErr`?